csclient/params/params.go

```diff
@@ -258,6 +262,30 @@ type WhoAmIResponse struct {
 	Groups []string
 }
 
+// Resource holds the metadata about a Resource defined for a charm.
+type Resource struct {
+	Name        string
+	Type        ResourceType
```

Comment:
Because it's basically free, and it makes the API much more descriptive.  If we don't define the acceptable values in the API, then consumers will have to look at the implementation of our code to figure out what the valid values are.